### PR TITLE
L1objects for heppy

### PR DESCRIPTION
**Pull request by silviodonato**
This code add the L1stage-2 objects in the VHbb ntuples (eg. L1 jets).

A few comments:
- At the moment, the L1 stage-2 objects are contained only in data and "reHLT" MC
- https://github.com/silviodonato/cmssw/blob/l1objects_Heppy/VHbbAnalysis/Heppy/test/vhbb.py#L396 needs to be overwritten with 'HLT2' in order to run with the reHLT samples 
- I needed to use an ugly repetitions of try/except in  https://github.com/silviodonato/cmssw/blob/l1objects_Heppy/PhysicsTools/Heppy/python/analyzers/core/L1TriggerAnalyzer.py#L45 in order to have a code working also with the no-reHLT MC
  - I tried to use the isValid() command, but it is always False until you call product()
  - I prefered to keep separated loops for each kind of object (jets,taus, EGammas, muons) 

I intend also to add two important features:
- add the L1 prescale information
- add more information about the object (eg. isolation for  EGamma objects).

### CODE REVIEW

**Comment by arizzi** [comment]
well I suggest you try only on the first one and then set a flag if it fails so that later code is just not run

**Comment by silviodonato** [comment]
I changed the code:
- now, it uses only one big try/except
- I defined the L1 candidate format (including quality and isolation, see [1]) and put it in the vhbb.py 

[1] https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/DataFormats/L1Trigger/src/L1Candidate.cc

File: PhysicsTools/Heppy/python/analyzers/core/L1TriggerAnalyzer.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        super(L1TriggerAnalyzer,self).beginLoop(setup)
+
+    def process(self, event):
+        self.readCollections( event.input )
+        event.l1Jets = []
+        event.l1Taus = []
+        event.l1Muons = []
+        event.l1EGammas = []
+        event.l1MET2,event.l1MET,event.l1ET,event.l1MHT,event.l1HT = tuple([TwoObjectsClass()]*5)
+        try:
```
Comment (arizzi) [inline PhysicsTools/Heppy/python/analyzers/core/L1TriggerAnalyzer.py]: if it fails at first event should it always fail? if so, can you add a flag in the except and avoid the "try" in later events?

**Comment by silviodonato** [comment]
I added the flag.
In the code I defined a dummy class "TwoObjectsClass". Do you know if it has been already defined somewhere in Heppy? It is simply a class compatible with https://github.com/vhbb/cmssw/blob/vhbbHeppy76X/PhysicsTools/Heppy/python/analyzers/objects/autophobj.py#L15

PS. I'm not able to add the "Heppy" flag to this PR. Could you do it?

File: PhysicsTools/Heppy/python/analyzers/core/L1TriggerAnalyzer.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import ROOT
+
+from PhysicsTools.Heppy.analyzers.core.Analyzer import Analyzer
+from PhysicsTools.Heppy.analyzers.core.AutoHandle import AutoHandle
+from PhysicsTools.HeppyCore.utils.deltar import matchObjectCollection, matchObjectCollection3
+import PhysicsTools.HeppyCore.framework.config as cfg
+
+
+class TwoObjectsClass():
```
Comment (arizzi) [inline PhysicsTools/Heppy/python/analyzers/core/L1TriggerAnalyzer.py]: please call it PtPhiPair or something like this

**Comment by silviodonato** [comment]
Dummy class renamed in PtPhiPairClass

File: VHbbAnalysis/Heppy/test/vhbb_combined_data.py [hunk 3]
```diff
@@ -19,7 +19,9 @@
 
 TriggerObjectsAna.triggerObjectInputTag = ('selectedPatTrigger','','RECO')
 FlagsAna.processName='RECO'
+TrigAna.processName = 'RECO'
```
Comment (arizzi) [inline VHbbAnalysis/Heppy/test/vhbb_combined_data.py]: why RECO? shouldn't it be HLT or HLT2 ?

**Comment by silviodonato** [comment]
It is for data.
They are stored under 'RECO' in /store/data/Run2016B/BTagCSV/MINIAOD/PromptReco-v2/000/273/537/00000/E8E0EF1B-7E1F-E611-8717-02163E012B71.root

**Comment by arizzi** [comment]
I doubt the HLT response is stored in RECO

**Comment by silviodonato** [comment]
My fault. I've fixed.

**Comment by silviodonato** [comment]
I think this PR would be useful to:
- study possible new L1 triggers (eg. L1MET+QCD veto for ZvvHbb)
- select only one L1 trigger in offline analysis (eg. select only L1_HTT trigger in HH->4b, in order to simplify the measurement of the efficiency)

Do you have any objection to integrate it?

**Comment by perrozzi** [comment]
this PR has conflicts: can you please solve?

**Comment by silviodonato** [comment]
conflict solved

**Comment by perrozzi** [comment]
thanks. shall I wait for any specific green light or the PR can be merged?

**Comment by silviodonato** [comment]
I think you can merge it. The "heppy" flag is just a reminder for later.

**Pull request merged**